### ISSUE TITLE
Incorrect usage of wcscpy_s

### ISSUE THREAD

**Issue by filipnavara** [dotnet/coreclr#22150]
https://lgtm.com/projects/g/dotnet/coreclr/alerts/?mode=tree&severity=error&ruleFocus=2152910697

```
wcscpy_s(strDefault, sizeof(strDefault), replacement);
--
  | This 'call to wcscpy_s' operation is limited to 8 bytes but the destination is only 4 bytes.
233 | strDefaultLength = replacementLength;
```

... and three others.

/cc @calumgrant

**Comment by RussKeldorph** [comment]
@wtgodbe @janvorli

**Comment by jkotas** [comment]
Fixed by #22153
